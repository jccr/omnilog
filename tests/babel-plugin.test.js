const pluginTester = require("babel-plugin-tester").default;
const babelPlugin = require("../babel-plugin");

pluginTester({
  pluginName: "babel-plugin",
  plugin: babelPlugin,
  filename: __filename,
  babelOptions: {
    plugins: ["@babel/plugin-syntax-jsx"],
  },
  tests: {
    "does not change code with no identifiers": 'alert("hello");',
    "changes this code": {
      code: 'console.log("hello");',
      output: 'omnilog.log("hello", "[unknown]:1:0")();',
    },
    "using fixtures file": {
      fixture: "__fixtures__/hello-world.jsx",
      snapshot: true,
    },
    "using jest snapshots": {
      code: `
        var foo = 1;
        if (foo) console.log(foo);

        console.log("hello world");

        console.log("hello world", "foo", "bar");

        console.log(foo ? "yes" : "no"); // comment
      `,
      snapshot: true,
    },
    "sets __OMNILOG_BABEL_PLUGIN__ from false to true": {
      code: "globalThis.omnilog.__OMNILOG_BABEL_PLUGIN__ = false;",
      output: "globalThis.omnilog.__OMNILOG_BABEL_PLUGIN__ = true;",
    },
    "sets __OMNILOG_BABEL_PLUGIN__ from falsy to true": {
      code: "omnilog.__OMNILOG_BABEL_PLUGIN__ = 0;",
      output: "omnilog.__OMNILOG_BABEL_PLUGIN__ = true;",
    },
    "keeps __OMNILOG_BABEL_PLUGIN__ as true": {
      code: "x.__OMNILOG_BABEL_PLUGIN__ = true;",
      output: "x.__OMNILOG_BABEL_PLUGIN__ = true;",
    },
  },
});
