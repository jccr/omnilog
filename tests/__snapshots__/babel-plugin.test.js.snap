// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin using fixtures file: using fixtures file 1`] = `

import React from "react";

const HelloWorld = () => {
  function sayHello() {
    console.log("Hello, World!");
    console.info("Hello, World!");
    console.warn("Hello, World!");
    console.error("Hello, World!");
    console.debug("Hello, World!");
    console.dir({ hello: "world" });
  }

  return <button onClick={sayHello}>Click me!</button>;
};

export default HelloWorld;

      ↓ ↓ ↓ ↓ ↓ ↓

import React from "react";

const HelloWorld = () => {
  function sayHello() {
    omnilog.log("Hello, World!", "/tests/__fixtures__/hello-world.jsx:5:4")();
    omnilog.info("Hello, World!", "/tests/__fixtures__/hello-world.jsx:6:4")();
    omnilog.warn("Hello, World!", "/tests/__fixtures__/hello-world.jsx:7:4")();
    omnilog.error("Hello, World!", "/tests/__fixtures__/hello-world.jsx:8:4")();
    omnilog.debug("Hello, World!", "/tests/__fixtures__/hello-world.jsx:9:4")();
    console.dir({
      hello: "world",
    });
  }

  return <button onClick={sayHello}>Click me!</button>;
};

export default HelloWorld;


`;

exports[`babel-plugin using jest snapshots: using jest snapshots 1`] = `

var foo = 1;
if (foo) console.log(foo);

console.log("hello world");

console.log("hello world", "foo", "bar");

console.log(foo ? "yes" : "no"); // comment

      ↓ ↓ ↓ ↓ ↓ ↓

var foo = 1;
if (foo) omnilog.log(foo, "[unknown]:2:9")();
omnilog.log("hello world", "[unknown]:4:0")();
omnilog.log("hello world", "foo", "bar", "[unknown]:6:0")();
omnilog.log(foo ? "yes" : "no", "[unknown]:8:0")(); // comment


`;
